specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkSecurityPerimeter.json

```diff
@@ -351,6 +852,261 @@
       },
       "description": "Properties of network security perimeter."
     },
+    "NspProfile": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/NspProfileProperties",
+          "description": "Properties of the network security perimeter profile"
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the profile resource that is unique within a perimeter. This name can be used to access the resource."
+        },
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Identifier of the network security perimeter profile in ARM id format."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/Resource"
+        }
+      ],
+      "description": "The network security perimeter profile resource"
+    },
+    "NspProfileListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NspProfile"
+          },
+          "description": "Gets a page of NSP profile"
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Gets the URL to get the next page of results."
+        }
+      },
+      "description": "Result of the request to list NSP profiles. Contains a list of NSP profiles and a URL link to get the next set of results."
+    },
+    "NspProfileProperties": {
+      "type": "object",
+      "properties": {
+        "accessRulesVersion": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Version number that increases with every update to access rules within the profile."
+        }
+      },
+      "description": "Properties of NSP profile."
+    },
+    "NspAccessRule": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/NspAccessRuleProperties",
+          "description": "Properties of the NSP access rule."
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the access rule that is unique within a profile. This name can be used to access the resource."
+        },
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "NSP access rule identifier."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/Resource"
```

Comment:
>Resource [](http://example.com/codeflow?start=47&length=8)

Same comment here as above.. is it expected to be unique within a resource group ?

Reply:
It is unique name in the RG of same type.

Reply:
It's unique name within the same profile. It has to be.

Reply:
Well not necessarily 

this is defined under : 
networkSecurityPerimeters/{networkSecurityPerimeterName}/resourceAssociations/{associationName}


so under a resource group  you can have 2 different nsp with same name for the associations .

again either ways is fine but I dont see why you would want to add such an artificial constraint on this